src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java

```diff
@@ -99,49 +108,138 @@ private GenerationModel mockGenerationModel(String example) {
     return flowEntry;
   }
 
+
+  @Test
+  public void blankDocumentWithExternalizedGlobals() {
+    String externalConfigurationFile = "globals.xml";
+    ScaffoldingAccessories scaffoldingAccessories =
+        new ScaffoldingAccessories(HIDE_CONSOLE, externalConfigurationFile, null, null);
+    ScaffoldingConfiguration.Builder builder =
+        ScaffoldingConfiguration.builder().withShowConsole(HIDE_CONSOLE).withExternalCommonFile(externalConfigurationFile);
+    List<MuleConfig> muleConfigs = scaffoldBlankDocument(builder.build());
+    for (MuleConfig muleConfig : muleConfigs) {
+      if (muleConfig.getName() == externalConfigurationFile) {
+        assertEquals(new ArrayList<>(), muleConfig.getApiAutodiscoveryConfig());
+        assertEquals(muleConfig.getHttpListenerConfigs().size(), 1);
+        assertEquals(muleConfig.getApikitConfigs().size(), 1);
+      } else {
+        assertEquals(muleConfig.getMainFlows().size(), 1);
+      }
+    }
+  }
+
+  @Test
+  public void blankDocumentWithoutExternalizedGlobals() {
+    ScaffoldingAccessories scaffoldingAccessories = new ScaffoldingAccessories(HIDE_CONSOLE, null, null, null);
+    ScaffoldingConfiguration.Builder builder =
+        ScaffoldingConfiguration.builder().withShowConsole(HIDE_CONSOLE);
+    List<MuleConfig> muleConfigs = scaffoldBlankDocument(builder.build());
+    for (MuleConfig muleConfig : muleConfigs) {
+      Document document = muleConfig.getContentAsDocument();
+      Element rootElement = document.getRootElement();
+      assertEquals("mule", rootElement.getName());
+      assertConfigurations(rootElement);
+    }
+  }
+
+  @Test
+  public void blankDocumentWithAPIAutodiscovery() {
+    String apiAutodiscovery = "1234";
+    ScaffoldingAccessories scaffoldingAccessories = new ScaffoldingAccessories(HIDE_CONSOLE, null, apiAutodiscovery, null);
+    ScaffoldingConfiguration.Builder builder =
+        ScaffoldingConfiguration.builder().withShowConsole(HIDE_CONSOLE).withApiId(apiAutodiscovery);
+    List<APIAutodiscoveryConfig> expectedApiAutodiscoveryConfig =
+        Arrays.asList(new APIAutodiscoveryConfig("1234", true, "hello-main"));
+    List<MuleConfig> muleConfigs = scaffoldBlankDocument(builder.build());
+    assertEquals(muleConfigs.size(), 1);
+    for (MuleConfig muleConfig : muleConfigs) {
+      assertEquals(expectedApiAutodiscoveryConfig, muleConfig.getApiAutodiscoveryConfig());
```

Comment:
You are comparing lists here. This is not correct. Please, update to assert the configuration details.